sympy/ntheory/generate.py

```diff
@@ -836,7 +842,7 @@ def randprime(a, b):
     if a >= b:
         return
     a, b = map(int, (a, b))
-    n = random.randint(a - 1, b)
+    n = randint(a - 1, b)
```

Comment:
```suggestion
    n = randint(a - 1, b)
    if n % 2:
        p = prevprime(n)
        if p >= a:
            return p
```
and below, `if p >= b and n%2 == 0:`